cmd/cloudinit/cloudinit.go

```diff
@@ -229,7 +230,9 @@ func saveCloudConfig() error {
 		}
 	}
 
-	userData := string(userDataBytes)
+	env := initialize.NewEnvironment("", "", "", "", metadata)
+	userData := env.Apply(string(userDataBytes))
+	userDataBytes = []byte(userData)
```

Comment:
as a private function in cloudinit.go: the function would then be called by `saveCloudConfig()` and, of course, by the test.